File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arr) {
        int m = arr.size();
        int hv = arr[0].back();
        int lv = arr[0][0];
        int ans = 0;
        
        for (int i=1; i<m; i++) {
            ans = max(ans, abs(arr[i].back() - lv));
            ans = max(ans, abs(hv - arr[i][0]));
            lv = min(lv, arr[i][0]);
            hv = max(hv, arr[i].back());
        }
        
        return ans;        
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arr: List[List[int]]) -> int:
        lv = arr[0][0]
        hv = arr[0][-1]
        ans = 0

        for i in range(1, len(arr)):
            ans = max(ans, hv - arr[i][0])
            ans = max(ans, arr[i][-1] - lv)
            hv = max(hv, arr[i][-1])
            lv = min(lv, arr[i][0])
        
        return ans